ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -0,0 +1,63 @@
+module XenAPI = Client.Client
+module Date = Xapi_stdext_date.Date
+
+let seconds_per_day = 3600. *. 24.
+let seconds_per_30_days = 30. *. seconds_per_day
+
+let days_until_expiry epoch expiry =
+  int_of_float (expiry /. seconds_per_day -. epoch /. seconds_per_day)
+
+let get_certificate_attributes rpc session =
+  XenAPI.Certificate.get_all_records rpc session
```

Comment:
Since this get the certs of all hosts in the pool, we need to make sure that the script runs on just one host (the master). I can't remember if/how the license-check does that.

Reply:
I haven't seen anything ensuring this on license-check, which is bizarre, as I would expect it to just execute on a single host.

Reply:
Let's update the cron script to only continue if the file `/etc/xensource/pool.conf` reads `master`.

Reply:
I've made it fail if it's not run on a master, it seems more informative than exiting with 1 or not doing anything silently. Being successful with an info message could be another option, but I'm not sure where would that be logged, if at all.

Reply:
IMO the program should always work, on any host. What we are constraining, is the scheduling by cron.